src/lsptoolshost/debugger/debugger.ts

```diff
@@ -51,9 +51,54 @@ export function registerDebugger(
     context.subscriptions.push(
         vscode.debug.registerDebugConfigurationProvider('coreclr', dotnetWorkspaceConfigurationProvider)
     );
+
     context.subscriptions.push(
-        vscode.commands.registerCommand('dotnet.generateAssets', async (selectedIndex) =>
-            generateAssets(workspaceInformationProvider, selectedIndex)
-        )
+        vscode.commands.registerCommand('dotnet.generateAssets', async (selectedIndex) => {
+            if (!(await promptForDevKitDebugConfigurations())) {
+                return;
+            }
+
+            await generateAssets(workspaceInformationProvider, selectedIndex);
+        })
     );
 }
+
+async function promptForDevKitDebugConfigurations(): Promise<boolean> {
+    if (getCSharpDevKit()) {
+        let result: boolean | undefined = undefined;
+
+        while (result === undefined) {
```

Comment:
as far as I know the while loop isn't needed - `await vscode.window.showInformationMessage` will block until either an option is selected or the dialog is dismissed.

Reply:
`await vscode.window.showInformationMessage` will block but the loop is specifically for if the user selected `More Information` and the dialog will just show up again.